packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1535 @@
+          final String line = lines[j];
+          if (j > 0)
+            builder.write(config.lineBreak);
+          builder.write(debugWordWrap(line, kWrapWidth, wrapIndent: '  ').join('\n'));
+        }
+      }
+      builder.write(config.lineBreak);
+    }
+    if (properties.isNotEmpty)
+      builder.write(config.afterProperties);
+
+    final String prefixChildren = '$prefixOtherLines${config.bodyIndent}';
+
+    if (children.isEmpty &&
+        config.addBlankLineIfNoChildren &&
+        builder.hasMultipleLines) {
+      final String prefix = prefixChildren.trimRight();
+      if (prefix.isNotEmpty)
+        builder.writeRaw('$prefix${config.lineBreak}');
+    }
+
+    if (children.isNotEmpty && config.showChildren) {
+      if (config.isBlankLineBetweenPropertiesAndChildren &&
+          properties.isNotEmpty &&
+          children.first.textTreeConfiguration.isBlankLineBetweenPropertiesAndChildren) {
+        builder.write(config.lineBreak);
+      }
+
+      for (int i = 0; i < children.length; i++) {
+        final DiagnosticsNode child = children[i];
+
+        final TextTreeConfiguration childConfig = _childTextConfiguration(child, config);
+        if (i == children.length - 1) {
+          final String lastChildPrefixLineOne =
+              '$prefixChildren${childConfig.prefixLastChildLineOne}';
+          if (child == null) {
+            builder.writeRawLine('$lastChildPrefixLineOne<null>');
+            continue;
+          }
+          builder.writeRawLine(child.toStringDeep(
+              lastChildPrefixLineOne,
+              '$prefixChildren${childConfig.childLinkSpace}${childConfig.prefixOtherLines}'));
+          if (childConfig.footer.isNotEmpty)
+            builder.writeRaw('$prefixChildren${childConfig.childLinkSpace}${childConfig.footer}');
+
+        } else {
+          final TextTreeConfiguration nextChildStyle =
+              _childTextConfiguration(children[i + 1], config);
+
+          final String childPrefixLineOne = '$prefixChildren${childConfig.prefixLineOne}';
+          final String childPrefixOtherLines ='$prefixChildren${nextChildStyle.linkCharacter}${childConfig.prefixOtherLines}';
+
+          if (child == null) {
+            builder.writeRawLine('$childPrefixLineOne<null>');
+            continue;
+          }
+          builder.writeRawLine(
+              child.toStringDeep(childPrefixLineOne, childPrefixOtherLines));
+          if (childConfig.footer.isNotEmpty)
+            builder.writeRaw('$prefixChildren${nextChildStyle.linkCharacter}${childConfig.footer}');
+        }
+      }
+    }
+    return builder.toString();
+  }
+}
+
+/// Property which supports specifying whether the string is value is enclosed
+/// in quotes when displayed.
+class StringProperty extends DiagnosticsProperty<String> {
+  StringProperty(String name, String value, {
+    String description,
+    bool showName: true,
+    Object defaultValue: kNoDefaultValue,
+    bool hidden: false,
+    this.quoted: false,
```

Comment:
Agreed. Set to true and moved most of the cases where it was false to a new MessageProperty class for messages that should be displayed like properties. It is possible that all the few remaining true string properties should be quoted as well and quotes can be required for String properties. Font is the string property where quotes would have the least clear benefit.